### PR TITLE
Improve getchaintxstats test coverage

### PR DESCRIPTION
**Pull request by promag**

### CODE REVIEW

File: test/functional/blockchain.py [hunk 1]
```diff
@@ -100,6 +100,21 @@ def _test_getblockchaininfo(self):
...
+        assert_raises_rpc_error(-1, "JSON value is not an integer as expected", self.nodes[0].getchaintxstats, '')
+        assert_raises_rpc_error(-8, "Invalid block count: should be between 0 and the block's height - 1", self.nodes[0].getchaintxstats, -1)
+        assert_raises_rpc_error(-8, "Invalid block count: should be between 0 and the block's height - 1", self.nodes[0].getchaintxstats, self.nodes[0].getblockcount())
+
+        # Test `getchaintxstats` invalid `blockhash`
+        assert_raises_rpc_error(-1, "JSON value is not a string as expected", self.nodes[0].getchaintxstats, blockhash=0)
+        assert_raises_rpc_error(-5, "Block not found", self.nodes[0].getchaintxstats, blockhash='0')
+        blockhash = self.nodes[0].generate(1)[0];
+        self.nodes[0].invalidateblock(blockhash)
+        assert_raises_rpc_error(-8, "Block is not in main chain", self.nodes[0].getchaintxstats, blockhash=blockhash)
```
Comment (ajtowns) [inline test/functional/blockchain.py]: Wouldn't it be better to do:

    blockhash = self.nodes[0].getblockhash(200)
    self.nodes[0].invalidateblock(blockhash)
    assert_raises_rpc_error()
    self.nodes[0].reconsiderblock(blockhash)

and drop the invalidateblock in _test_gettxoutsetinfo()?

**Review by ajtowns — APPROVED** [review]
ACK 5b95f96871a33aada918c54c4eff263cb3c55fec -- refactoring looks like it shouldn't change the logic; extra test cases look good.

File: test/functional/blockchain.py (hunk 1 above)
Comment (promag) [reply test/functional/blockchain.py]: Done, I'll squash once it's green.

**Comment by promag** [comment]
@ajtowns Thanks for the review.

> refactoring looks like it shouldn't change the logic

If this is a concern for others, I take that commit to a new PR.

**Review by ryanofsky — COMMENTED** [review]
utACK 590aa8f1b56cf1a8d82bff41a513cf5f3f530e4a

**Comment by promag** [comment]
Rebased to fix conflict of test file `test/functional/rpc_blockchain.py` rename.

Best reviewed with `?w=1`.

**Pull request merged**